validation/src/test/scala/hmda/validation/rules/lar/quality/Q027Spec.scala

```diff
@@ -0,0 +1,42 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+
+class Q027Spec extends LarEditCheckSpec {
+  property("fails when actionTaken = 1,2,3,4,5,7,8 and propertyType = 1,2 and income NA") {
+    forAll(larGen) { lar =>
+      whenever(lar.actionTakenType != 6 && lar.loan.propertyType != 3) {
+        val invalidApplicant = lar.applicant.copy(income = "NA")
+        val invalidLar = lar.copy(applicant = invalidApplicant)
+        invalidLar.mustFail
+      }
+    }
+  }
+
+  property("passes when applicant income not NA") {
+    forAll(larGen) { lar =>
+      whenever(lar.applicant.income != "NA") {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("passes when action taken not 1,2,3,4,5,7,8") {
+    forAll(larGen) { lar =>
+      val validLar = lar.copy(actionTakenType = 6)
```

Comment:
@schbetsy @kgudel It is unlikely that the valid values will change but we have the ability to change them by taking the data spec out of the reg so we cannot rule that out. You can probably leave as-is with the caveat (or team understanding) that if valid values ever change, the team needs to go back and update tests like this to be correct for the new valid values. Hope this helps!